python/python/lancedb/query.py

```diff
@@ -620,7 +623,32 @@ def metric(self, metric: Literal["L2", "cosine", "dot"]) -> LanceVectorQueryBuil
         LanceVectorQueryBuilder
             The LanceQueryBuilder object.
         """
-        self._metric = metric.lower()
+        return self.distance_type(metric)
+
+    def distance_type(
+        self, distance_type: Literal["L2", "cosine", "dot"]
+    ) -> "LanceVectorQueryBuilder":
+        """Set the distance metric to use.
+
+        When performing a vector search we try and find the "nearest" vectors according
+        to some kind of distance metric. This parameter controls which distance metric
+        to use.
+
+        Note: if there is a vector index then the distance type used MUST match the
+        distance type used to train the vector index. If this is not done then the
+        results will be invalid.
+
+        Parameters
+        ----------
+        distance_type: "L2" or "cosine" or "dot"
+            The distance metric to use. By default "L2" is used.
+
+        Returns
+        -------
+        LanceVectorQueryBuilder
+            The LanceQueryBuilder object.
+        """
+        self._metric = distance_type.lower()
```

Comment:
It would be nice to change the internal name to `distance_type` as well:

```suggestion
        self._distance_type = distance_type.lower()
```

(This will require other changes as well.)

Reply:
Hi Will, I have made the appropriate changes. Kindly, do have a look again.